src/Integration/Binding/BindingController.cs

```diff
@@ -147,18 +147,21 @@ private void OnBindingStarted()
         private void OnBindingFinished(ProjectInformation projectInformation, bool isFinishedSuccessfully)
         {
             this.IsBindingInProgress = false;
-            this.host.VisualStateManager.ClearBoundProject();
 
             if (isFinishedSuccessfully)
             {
-                this.host.VisualStateManager.SetBoundProject(projectInformation);
+                if (projectInformation.Key != this.host.VisualStateManager.BoundProjectKey)
```

Comment:
Not sure if we need this optimization, The bound project key might be the same but the actual bound projects may not be. Consider the case when you added a project to a bound solution, close VS, and the re-opened. SVS-9 will show an info bar and will allow the user to re-bind which means, for the unbound project, that it is now in connected mode.

Let's avoid this optimization for now, until we have a concrete perf issue at which point we will need to think holistically and might do something else entirety.